source/val/validate_memory.cpp

```diff
@@ -325,6 +325,17 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
     }
   }
 
+  // Check that pointer type and variable type have the same storage class
+  const auto result_storage_class_index = 1;
+  const auto result_storage_class =
+      result_type->GetOperandAs<uint32_t>(result_storage_class_index);
```

Comment:
Minor: template operand should be SpvStorageClass. Please update `storage_class` above too.

Reply:
Done